js/index.js

```diff
@@ -1227,3 +1227,47 @@ if (btn) btn.addEventListener("click", event => {
 
   window.open(url, '_blank');
 });
+
+btn = document.getElementById("screenshot");
+if (btn) btn.addEventListener("click",event=>{
+  let url;
+  Progress.show({title:"Creating screenshot",interval:10,percent:"animate",sticky:true});
+  Comms.write("\x10g.dump();\n").then((s)=>{
+    let oImage = new Image();
+    oImage.onload = function(){
+      Progress.show({title:"Converting screenshot",percent:90,sticky:true});
+      let oCanvas = document.createElement('canvas');
+      oCanvas.width = oImage.width;
+      oCanvas.height = oImage.height;
+      let oCtx = oCanvas.getContext('2d');
+      oCtx.drawImage(oImage, 0, 0);
+      url = oCanvas.toDataURL();
+
+      let screenshotHtml = `
+        <div style="text-align: center;">
+          <img align="center" src="${url}"></img>
+        </div>
+      `
+
+      showPrompt("Save Screenshot?",screenshotHtml, undefined, false).then((r)=>{
+        Progress.show({title:"Saving screenshot",percent:99,sticky:true});
+        let link = document.createElement("a");
+        link.download = "screenshot.png";
+        link.target = "_blank";
+        link.href = url;
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+      }).catch(()=>{
+      }).finally(()=>{
+        Progress.hide({sticky:true});
+      });
+    }
+    oImage.src = s.split("\n")[0];
+    Progress.hide({sticky:true});
+    Progress.show({title:"Screenshot done",percent:85,sticky:true});
+
+  }, err=>{
+    showToast("Error creating screenshot: "+err,"error");
```

Comment:
Nice! Does this need a `Progress.hide(...)` call here?

Reply:
Do you mean line 1267? Without that it does not change from animated progress bar to fixed percentages. There might be a better way to do this.

Reply:
Oh no, I was thinking if we hit an error we could've done `Progress.show({title:"Creating screenshot",interval:10,percent:"animate",sticky:true});`, and then no other change to `Progress` in this error handler section, if you see what I mean?